### ISSUE TITLE
CodeBlock component that accepts language and code props

### ISSUE THREAD

**Issue by xaviergonz** [doczjs/docz#1295]
Basically I need to print a string of code as if it were a (typescript / tsx) code block, but code blocks don't support passing dynamic strings.

Would it be possible to expose a component such as:
```
<CodeBlock code={myString} language="tsx"/>
```
that would "kind of" do the same as:
```
  ` ` ` tsx
    {myString}
  ` ` ` 
```

The point is that I'm loading a code from a file using raw-loader and would like to show it. In V1 I got it working with
```
  <editor>{myString}</editor>
```
but that no longer works

**Comment by rakannimer** [comment]
Hello @xaviergonz,

You can use docz's code block component by importing it from the `gatsby-docz-theme` : 

```jsx
import { Code } from 'gatsby-theme-docz/src/components/Code'

<Code className={'tsx'}>
  {`
interface IState {
  clicks: number
}
return () => <App />
 `}
</Code>

```

However when testing this code before writing this I realized that version 2.1.1 of docz depends on an older version of prism-react-renderer that breaks when parsing tsx (no syntax highlighting). 

This issue will be fixed in 2.2.0 and you can try it out now by using `2.2.0-alpha.5`.

```sh
yarn add docz@next
```

**Comment by xaviergonz** [comment]
Thanks, that worked. Btw, shouldn't the generated pre element have overflow-x: auto? right now if a line of code is too long it will just overflow outside of the code box (easy to happen in mobile)

**Comment by rakannimer** [comment]
> shouldn't the generated pre element have overflow-x: auto? right now if a line of code is too long it will just overflow outside of the code box

Yep thanks for catching that, it was not ideal. Fixed it in `2.2.0-alpha.6` 👍
